hwy/ops/x86_128-inl.h

```diff
@@ -870,6 +870,19 @@ HWY_API MFromD<D> MaskFalse(D /*d*/) {
   return MFromD<D>{static_cast<decltype(MFromD<D>().raw)>(0)};
 }
 
+// ------------------------------ IsNegative (MFromD)
+#ifdef HWY_NATIVE_IS_NEGATIVE
+#undef HWY_NATIVE_IS_NEGATIVE
+#else
+#define HWY_NATIVE_IS_NEGATIVE
+#endif
+
+// Generic for all vector lengths
+template <class V, HWY_IF_NOT_UNSIGNED_V(V)>
+HWY_API MFromD<DFromV<V>> IsNegative(V v) {
+  return MaskFromVec(v);
```

Comment:
I was concerned about this one because it's not obvious from the diff that we are in the AVX3 section, where this is safe. I've verified that we are, so all good. But consider adding a comment that this works (only) for AVX3+, whereas on AVX2 we have to return 0xFF..FF in masks?